### PR TITLE
fix(essntl-4735): fix for the tags modal filter (filter case sensitivity)

### PR DESCRIPTION
**Pull request by Fewwy**
I added a filter based on regular expressions that is case insensitive and have improved performance over a .filter method 

Link to the card
https://issues.redhat.com/browse/ESSNTL-4735

Screenshots
image

### CODE REVIEW

File: src/Utilities/regExpInsensitiveFilter.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+const toReg = str =>
+    new RegExp(str?.replace(/\//g, '//'), 'gi');
+
+// test a string with a regular expression
+const match = reg => x =>
+    reg?.test(x);
+
+// filter an array by a predicate
+// const filter = f => a => a.filter(a)
+const filter = f => a => {
```
Comment (bastilian) [inline src/Utilities/regExpInsensitiveFilter.js:25]: .... is this reimplementing `Array.filter`?

File: src/Utilities/regExpInsensitiveFilter.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+// run a predicate function over each key of an object
+// const hasValue = f => o =>
+//  Object.keys(o).some(x => f(o[x]))
+const hasValue = f => o => {
+    let key;
+    for (key in o) {
+        if (f(o[key])) {return true;}
+
+        ;
```
Comment (bastilian) [inline src/Utilities/regExpInsensitiveFilter.js:9]: ?

**Comment by bastilian** [comment]
@Fewwy This seems to be filtering only on the client, but it's not querying the API, or is it?

**Comment by Fewwy** [comment]
@bastilian Yes, it fetches new list only when you open the Tags Modal

File: src/Utilities/regExpInsensitiveFilter.js (hunk 2 above)
Comment (Fewwy) [reply src/Utilities/regExpInsensitiveFilter.js:9]: This assert's that the object has the value we need

**Comment by bastilian** [comment]
@Fewwy But what if there are more tags to fetch that match the searched term?

**Comment by Fewwy** [comment]
@bastilian I don't think this PR should add more fetching to the tags modal. This is a bug fix PR
If we want to add more fetching depending on the filter - that's an update of the existing feature and we have to ask a PM for that.

File: src/Utilities/regExpInsensitiveFilter.js (hunk 2 above)
Comment (bastilian) [reply src/Utilities/regExpInsensitiveFilter.js:9]: Why is the `;` flying down here?

File: src/Utilities/regExpInsensitiveFilter.js (hunk 1 above)
Comment (bastilian) [reply src/Utilities/regExpInsensitiveFilter.js:25]: Please, do not re-implement a native function. It's doubtful that it is more performant than a native implementation, even if it is more performant it's probably not worth the confusion it contributes.

File: src/Utilities/regExpInsensitiveFilter.js (hunk 2 above)
Comment (Fewwy) [reply src/Utilities/regExpInsensitiveFilter.js]: Ah, I thought you are asking about the function itself. I removed these floating ;

File: src/Utilities/regExpInsensitiveFilter.js (hunk 1 above)
Comment (Fewwy) [reply src/Utilities/regExpInsensitiveFilter.js]: @bastilian I rewrote the function

File: src/Utilities/TagsModal.js [hunk 3]
```diff
@@ -36,10 +37,10 @@ const TagsModal = ({
     const activeSystemTag = useSelector(({ entities, entityDetails }) => entities?.activeSystemTag || entityDetails?.entity);
     const tags = useSelector(({ entities, entityDetails }) => {
         const activeTags = entities?.activeSystemTag?.tags || entityDetails?.entity?.tags;
+
         if (activeTags) {
-            return activeTags?.filter(
-                (tag) => Object.values(tag).some((val) => val?.includes(filterBy))
```
Comment (bastilian) [inline src/Utilities/TagsModal.js:41]: Why not just 
```js
val?.toLowerCase().includes(filterBy.toLowerCase())
```

? Does case-insensitive filtering and is far simpler.

File: src/store/inventory-actions.js [hunk 4]
```diff
@@ -170,11 +170,13 @@ export const toggleTagModal = (isOpen) => ({
     payload: { isOpen }
 });
 
-export const fetchAllTags = (search, pagination, getTags = defaultGetAllTags) => ({
-    type: ACTION_TYPES.ALL_TAGS,
-    payload: getTags(search, pagination),
-    meta: { lastDateRequestTags: Date.now() }
-});
+export const fetchAllTags = (search, pagination, getTags = defaultGetAllTags) => {
+    return {
```
Comment (bastilian) [inline src/store/inventory-actions.js:174]: We should revert this to the implicit return as it was before to keep the same form as the other actions.

File: src/Utilities/TagsModal.js (hunk 3 above)
Comment (Fewwy) [reply src/Utilities/TagsModal.js:41]: Hm, you are right on that point, I tried to do that in the first place but couldn't make it to work for me

File: src/store/inventory-actions.js (hunk 4 above)
Comment (Fewwy) [reply src/store/inventory-actions.js:174]: Yep, that's a clean up I forgot to do, was checking the values of these parameters and forgot to change it back

File: src/Utilities/TagsModal.js (hunk 3 above)
Comment (bastilian) [reply src/Utilities/TagsModal.js:41]: This is pretty much how all (client side) filtering in Compliance works for string searches and it works fine there.

**Comment by gkarat** [comment]
This merge with master ^ should fix the failing cy test not related to this PR.

**Review by bastilian — APPROVED** [review]
LGTM! Thank you @Fewwy!

**Pull request merged**